src/bp/core/services/telemetry/sdk-methods.ts

```diff
@@ -0,0 +1,185 @@
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository,
+    @inject(TYPES.BotService) private botService: BotService
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-SDK'
+    this.interval = ms('1d')
+  }
+
+  protected async getStats() {
+    return {
+      ...buildSchema(await this.getServerStats(), 'server'),
+      event_type: 'custom_roles',
+      event_data: { schema: '1.0.0', SDKMethods: await this.SDKUsage() }
+    }
+  }
+
+  private async SDKUsage(): Promise<SDKUsageEvent> {
+    const bots = await this.botService.getBotsIds()
+    return { actions: await this.getActionUsages(bots), hooks: await this.getHooksUsages() }
+  }
+
+  private async getActionUsages(bots: string[]): Promise<SDKUsage> {
+    const rootFolder = 'actions'
+    const globalActionsNames = await this.ghostService
+      .global()
+      .directoryListing('/', `${rootFolder}/*.js`)
+      .map(path => path.split('/').pop() || '')
+
+    const global = await Promise.map(globalActionsNames, this.parseFile(`/${rootFolder}`))
+    const perBots = await Promise.reduce(bots, this.botActionsReducer(rootFolder), [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private botActionsReducer(rootFolder: string): (acc: any, botId: any) => Promise<ParsedFile[]> {
+    return async (acc, botId) => {
+      const botActionsNames = await this.ghostService.forBot(botId).directoryListing(`/${rootFolder}`, '*.js')
+      const parsedFiles = await Promise.map(botActionsNames, this.parseFile(`/${rootFolder}`, { botId }))
+      return [...acc, ...parsedFiles]
+    }
+  }
+
+  private async getHooksUsages(): Promise<SDKUsage> {
+    const hooksPayload = await getHooksLifecycle(this.botService, this.ghostService)
+    const global = await this.parseHooks(hooksPayload.global.filter(hook => hook.type === 'custom'))
+    const perBots = await Promise.reduce(hooksPayload.perBots, this.botHooksReducer(), [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private botHooksReducer(): (acc: any, botHooks: BotHooks) => Promise<ParsedFile[]> {
+    return async (acc, botHooks: BotHooks) => {
+      const { botId, hooks } = botHooks
+      const parsedFiles = await this.parseHooks(
+        hooks.filter(hook => hook.type === 'custom'),
+        botId
+      )
+      return [...acc, ...parsedFiles]
+    }
+  }
+
+  private async parseHooks(hooks: Hook[], botId?: string): Promise<ParsedFile[]> {
+    return Promise.map(hooks, async hook => {
+      const { lifecycle, enabled, path } = hook
+      const usageParams = { lifecycle, enabled }
+
+      if (botId) {
+        _.set(usageParams, 'botId', botId)
+      }
+
+      return this.parseFile('/hooks', usageParams)(path!)
+    })
+  }
+
+  private parseFile(rootFolder: string, usageParams?): (name: string) => Promise<ParsedFile> {
+    return async (name: string) => {
```

Comment:
Personnal pref, but I feel like there's no need for function currying here. Just pass all params at once:

```
private parseFile(rootFolder: string, name: string, usageParams: type): Promise<ParsedFile> {
...
}
```

Reply:
Agreed. I don't really like those methods which only return other methods... Harder to read.

Reply:
fixed: da361adc4dbd460510834ed4d8b804e2b1231fad